File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights) -1 
        ans = 0
        while(left < right) :
            ans = max(ans, 
                abs(right-left) * int(min(heights[left], heights[right])))
            if(heights[left] >= heights[right]):
                right -= 1
            else:
                left += 1
                
        return ans

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s (nums.begin(), nums.end());
        int ans = 0;

        for(int i = 0; i < nums.size(); ++i) {
            if(s.find(nums[i]-1) != s.end()) continue;
            int len = 0;
            int val = nums[i];
            while(s.find(val++) != s.end()) len++;
            ans = max(len, ans);
        }
        return ans;
    }
};
